grammar.js

```diff
@@ -167,68 +238,14 @@ module.exports = grammar({
     startinsert_statement: ($) => maybe_bang($, keyword($, "startinsert")),
     stopinsert_statement: ($) => keyword($, "stopinsert"),
 
-    encoding: ($) =>
-      choice(
-        "latin1",
-        "iso",
-        "koi8",
-        "koi8",
-        "macroman",
-        "cp437",
-        "cp737",
-        "cp775",
-        "cp850",
-        "cp852",
-        "cp855",
-        "cp857",
-        "cp860",
-        "cp861",
-        "cp862",
-        "cp863",
-        "cp865",
-        "cp866",
-        "cp869",
-        "cp874",
-        "cp1250",
-        "cp1251",
-        "cp1253",
-        "cp1254",
-        "cp1255",
-        "cp1256",
-        "cp1257",
-        "cp1258",
-        "cp932",
-        "euc-jp",
-        "sjis",
-        "cp949",
-        "euc-kr",
-        "cp936",
-        "euc-cn",
-        "cp950",
-        "big5",
-        "euc-tw",
-        "utf-8",
-        "ucs-2",
-        "ucs-21e",
-        "utf-16",
-        "utf-16le",
-        "ucs-4",
-        "ucs-4le",
-        "ansi",
-        "japan",
-        "korea",
-        "prc",
-        "chinese",
-        "taiwan",
-        "utf8",
-        "unicode",
-        "ucs2be",
-        "ucs-2be",
-        "ucs-4be",
-        "utf-32",
-        "utf-32le",
-        "default"
-      ),
+    file_format: ($) => choice_arr(FILE_FORMAT),
+    _immediate_file_format: ($) =>
+      alias(choice_arr(FILE_FORMAT, (is_immediate = true)), $.file_format),
+
+    encoding: ($) => choice(...ENCODING),
+    _immediate_encoding: ($) =>
+      alias(choice(...ENCODING.map(token.immediate)), $.encoding),
```

Comment:
I would like to know if this is ok for you, and if you prefert to use a function for that or juste use as for encoding